08_voting/index.tex

```diff
@@ -1 +1,192 @@
+
+\begin{figure}[H]
+\begin{center}
+\begin{tikzpicture}[node distance=20pt]
+\centering
+\node[draw, rounded corners] (start)  {The President selects a rule to be voted on};
+\node[draw, below=of start] (step 1)  {The President notifies the Speaker to hold a voting session on the selected rule};
+\node[draw, below=of step 1] (step 2)  {The Speaker opens ballots for all islands to cast the votes};
+\node[draw, below=of step 2] (step 3)  {The Speaker closes ballots after receiving votes from all islands};
+\node[draw, below=of step 3] (step 4)  {The Speaker counts the votes by calling a voting method/function for winner selection};
+\node[draw, below=of step 4, rounded corners] (end)   {The Speaker publicly declares the result of voting};
+ \draw[->] (start)  -- (step 1);
+ \draw[->] (step 1) -- (step 2);
+ \draw[->] (step 2) -- (step 3);
+ \draw[->] (step 3) -- (step 4);
+ \draw[->] (step 4) -- (end);
+\end{tikzpicture}
+\caption{Voting Protocol for Rules}
+\label{fig:RONRVotingProtocol}
+\end{center}
+\end{figure}
+
+For elections of roles, the sequence of actions of the voting protocol is mostly similar to the above explanation in principle, except for some parameters, such as the motion of the vote which is the role itself (President, or Speaker, or Judge), the facilitator of the election/vote event which depends on what role is being held for election (refer to Chapter 5 IIGO for more details on change of roles and power transfer), and the applicable voting method function to call for election that will produce the result, which is different from the voting method used for rules. Refer to Section~\ref{sec:VotingMethods} for more details on voting methods to be used for elections of roles.
+
+In the initial implementation, it is assumed that all 6 islands have the power to vote at any necessary voting scenario and no sanction or power restriction applied to any island when it comes to the right to vote. However, in the final implementation, some complications or dilemma are applied to, i.e. an/some island(s) could lose their right to vote or not permitted to participate in a voting event due to diplomatic sanction. Refer to Chapter 5 IIGO for details on diplomatic sanction. In this case, the Speaker will check and open the ballots to only eligible islands that have the right to vote at that certain stage, rather than to all islands as mentioned in the first scenario above.
```

Comment:
comma issue!

suggested-rewriting

In the initial turn of the game, it is assumed that all 6 islands have the power to vote at any necessary voting scenario**,** and **no diplomatic sanction is in place for any island.** **However, as in the further turns of the game, some island(s) could lose their right to vote and/or not permitted to participate in a voting event due to the diplomatic sanction(s) in place**. Refer to Chapter 5 IIGO for details on diplomatic sanction. **In this case, the Speaker will open the ballots to the eligible islands only** , **i.e. those who have are permitted to vote at a certain state of the game.**

Reply:
changed